test/extended/pods/oom_killer.go

```diff
@@ -0,0 +1,111 @@
+				Name:      podName,
+				Namespace: ns,
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{{
+					Name:  containerName,
+					Image: "k8s.gcr.io/stress:v1",
+					Args:  []string{"--mem-alloc-size", "5Gi", "--mem-total", "25Gi"},
+				},
+				},
+				PriorityClassName: "system-cluster-critical",
+			},
+		}
+		pod = podClient.Create(pod)
+		err := e2epod.WaitForPodNameRunningInNamespace(oc.KubeFramework().ClientSet, pod.Name, oc.Namespace())
+		framework.ExpectNoError(err)
+
+		pod, err = podClient.Get(context.TODO(), pod.Name, metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		defer func() {
+			g.By("deleting the pod")
+			podClient.Delete(context.TODO(), pod.Name, *metav1.NewDeleteOptions(0))
+		}()
+
+		// To wait for pod to start hogging memory
+		time.Sleep(2 * time.Minute)
+		oomEventMsgList := []string{}
+
+		// Collect the events to see if we're getting OOM Killed.
+		kubeClient := oc.AdminKubeClient()
+
+		events, err := kubeClient.CoreV1().Events("").List(context.TODO(), metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		containerOOMKilledCount := 0 // We need to search for container exited with code 137
+		systemOOMErrorMsg := "System OOM encountered"
+		containerOOMErrorMsg := "container exited with code 137 (OOMKilled)"
+		for _, event := range events.Items {
+			if strings.Contains(event.Message, containerOOMErrorMsg) && event.LastTimestamp.Time.After(currentTime) {
+				containerOOMKilledCount++
+				oomEventMsgList = append(oomEventMsgList, event.Message)
+			} else if strings.Contains(event.Message, systemOOMErrorMsg) && event.LastTimestamp.Time.After(currentTime) {
+				oomEventMsgList = append(oomEventMsgList, event.Message)
+			}
+		}
+		if len(oomEventMsgList) == 0 {
+			g.Fail(fmt.Sprint("Expected atleast one OOM event but did not get one"))
+		}
+		inducedOOMEventCount := 0
+		// The event should point to the pod that induced memory pressure
+		for _, oomEventMsg := range oomEventMsgList {
+			if strings.Contains(oomEventMsg, "stress") {
+				inducedOOMEventCount++
+			}
+		}
+		if inducedOOMEventCount != len(oomEventMsgList) {
+			result.Flakef("From events: expected oom inducing pod to be oomkilled but other pods got killed as "+
+				"well, OOM inducing pod killed %v times, total OOM kills encountered %v",
+				inducedOOMEventCount, len(oomEventMsgList))
+		}
+
+		testDuration := fmt.Sprintf("%v", exutil.DurationSinceStartInSeconds().Seconds()) + "s"
+		out, err := oc.Run("adm", "node-logs").Args(pod.Spec.NodeName, "--path=journal",
+			"--case-sensitive=false", "--grep=invoked oom-killer", "--since=-"+testDuration).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		hasStressPodBeenOOMKilledFromNodeLogs := false
+		if strings.Contains(strings.ToLower(out), "stress") {
+			hasStressPodBeenOOMKilledFromNodeLogs = true
+		}
+		if !hasStressPodBeenOOMKilledFromNodeLogs {
+			result.Flakef("From Nodelogs: expected oom inducing pod to be oomkilled but other pod oom killed")
```

Comment:
You can only invoke one Flakef per test, so you need to accumulate the messages that you want to print and then summarize them at the end.

Reply:
I see. I will change it then. Another thing, I was talking to @rphillips about this, he told getting node-logs may not be helpful at all if the node is already hung(because of soft lock in kernel).